src/stream/src/executor/hash_join.rs

```diff
@@ -1466,22 +1458,31 @@ mod tests {
             0,
         )
         .await;
-        let schema_len = match T {
-            JoinType::LeftSemi | JoinType::LeftAnti => source_l.schema().len(),
-            JoinType::RightSemi | JoinType::RightAnti => source_r.schema().len(),
-            _ => source_l.schema().len() + source_r.schema().len(),
+
+        let schema = match T {
+            JoinType::LeftSemi | JoinType::LeftAnti => source_l.schema().clone(),
+            JoinType::RightSemi | JoinType::RightAnti => source_r.schema().clone(),
+            _ => [source_l.schema().fields(), source_r.schema().fields()]
+                .concat()
+                .into_iter()
+                .collect(),
+        };
+        let schema_len = schema.len();
+        let info = ExecutorInfo {
+            schema,
+            pk_indices: vec![1],
+            identity: "HashJoinExecutor".to_string(),
         };
 
         let executor = HashJoinExecutor::<Key128, MemoryStateStore, T>::new(
             ActorContext::create(123),
+            info,
             Box::new(source_l),
             Box::new(source_r),
             params_l,
             params_r,
             vec![false],
-            vec![1],
             (0..schema_len).collect_vec(),
-            1,
             cond,
             vec![],
             "HashJoinExecutor".to_string(),
```

Comment:
Seems no need this

Reply:
Indeed. `op_info` was intended to be a replacement for `identity`, but since we now pass `identity` via `ExecutorInfo` to executors, we can put whatever info we want to `identity` now. Pushed a new commit to remove it from `HashJoinExecutor`.